### ISSUE TITLE
Custom RMSE loss in tutorial get difference tree structure with the original RMSE loss!!!

### ISSUE THREAD

**Issue by ccylance** [catboost/catboost#2646]
Problem:
I am trying to learn how to custom loss function and metric. I referred to the tutorial and used RMSE loss, but found that the leaf value is different from the integrated RMSE loss in the model.

```

class RmseObjective(object):
    def calc_ders_range(self, approxes, targets, weights):
        assert len(approxes) == len(targets)
        if weights is not None:
            assert len(weights) == len(approxes)
        
        result = []
        for index in range(len(targets)):
            der1 = targets[index] - approxes[index]
            der2 = -1

            if weights is not None:
                der1 *= weights[index]
                der2 *= weights[index]

            result.append((der1, der2))
        return result
class RmseMetric(object):
    def get_final_error(self, error, weight):
        return np.sqrt(error / (weight + 1e-38))

    def is_max_optimal(self):
        return False

    def evaluate(self, approxes, target, weight):
        assert len(approxes) == 1
        assert len(target) == len(approxes[0])

        approx = approxes[0]

        error_sum = 0.0
        weight_sum = 0.0

        for i in range(len(approx)):
            w = 1.0 if weight is None else weight[i]
            weight_sum += w
            error_sum += w * ((approx[i] - target[i])**2)

        return error_sum, weight_sum

x = np.array([ 10.40625   , 11.375     ,  7.75      , 10.65      , 11.89285714,
        5.8125    ,  8.25      ,  8.875     , 11.625     , 11.625     ,
        7.60714286,  4.125     ,  2.75      , 13.3125    , 12.75      ,
        8.53125 ])
y = np.array([0.58201677, 0.89175633, 0.6678255 , 0.23394588, 0.21943126,
       0.34399695, 0.04934427, 0.84941853, 0.86018197, 0.7264951 ,
       0.17764019, 0.66359799, 0.47518967, 0.96162571, 0.98941818,
       0.19111766])

model = CatBoostRegressor(
                        iterations=1, depth = 1, boosting_type='Ordered',
                        has_time = True,
                        leaf_estimation_method='Gradient',
                        l2_leaf_reg= 0,
                        learning_rate=1,
                        loss_function=RmseObjective(),
                        eval_metric=RmseMetric(),
                        score_function = "L2",
                        bootstrap_type = "No",
                        # loss_function='RMSE', eval_metric='RMSE',
                        )
pools = Pool(x, y)
model.fit(pools, silent=True)
model.plot_tree(tree_idx=0, pool=pools)
```
here is the result for custom RMSE loss
![image](https://github.com/catboost/catboost/assets/54780246/421dffae-3681-429b-ab7f-5723a520b74d)
This is the result for integrated RMSE loss 
![image](https://github.com/catboost/catboost/assets/54780246/01c83ec5-bad8-454a-b108-d65e4c2b7a50)

catboost version: catboost 1.2
Operating System: macos 14.4.1
CPU: M1
GPU: no

**Comment by ek-ak** [comment]
Hello!
The difference is that the CatBoost RMSE metric is worked with boost_from_average=True, while custom metrics do not work with boost_from_average at all. If you add 'boost_from_average=False' to your model and values will be exactly the same.

**Comment by ccylance** [comment]
> Hello! The difference is that the CatBoost RMSE metric is worked with boost_from_average=True, while custom metrics do not work with boost_from_average at all. If you add 'boost_from_average=False' to your model and values will be exactly the same.

@ek-ak Could you explain logic of `boost_from_average` in more detail? Does it mean the initial prediction is set to average of the target rather than 0 when `boost_from_average=True`
